exporter/zipkin/zipkin.go

```diff
@@ -50,6 +54,36 @@ func NewExporter(reporter reporter.Reporter, localEndpoint *model.Endpoint) *Exp
 	}
 }
 
+type Options struct {
+	// EndpointHostPort is the host:port combination for the Zipkin service
+	// e.g. 192.168.0.2:8080
+	EndpointHostPort string
+
+	// ReporterURI is the HTTP endpoint to which spans will be sent
+	// e.g. http://localhost:9411/api/v2/spans
+	ReporterURI string
+
+	// ReporterOptions are optional Options
+	// that modify the zipkin http Reporter.
+	ReporterOptions []ziphttp.ReporterOption
+}
+
+func NewExporterWithOptions(o *Options) (*Exporter, error) {
+	if o == nil {
+		// TODO: Perhaps fill in the defaults that OpenZipkin uses
+		o = new(Options)
+	}
+	if _, err := url.Parse(o.ReporterURI); err != nil {
+		return nil, fmt.Errorf("Failed to parse ReporterURI err: %v", err)
+	}
+	reporter := ziphttp.NewReporter(o.ReporterURI, o.ReporterOptions...)
```

Comment:
what alternative protocols (besides HTTP) are supported? should we assert that the passed-in URL protocol is HTTP before constructing an HTTP reporter? that way in future we can just add support for new protocols by accepting more protocols in the ReporterURI

Reply:
Am not sure about this to be honest, I just figured that if they want to create a non-HTTP based zipkin reporter, they can use `zipkin.NewReporter`. @rakyll this is perhaps another reason why we should keep the old style `zipkin.NewReporter` and the new constructor as `zipkin.NewReporterWithOptions`

Reply:
This makes me question why we are making this change. We are providing two different APIs to achieve the same functionality -- confusing. I prefer we keep this exporter the way it is. Even though it is tedious, people will just copy paste the example. I don't think it is a big deal.

Reply:
@rakyll the point of this PR is to cut out the tediousness, I find it quite odd that in our example code we are also setting up boilerplate for the user. I have never seen that. I started with NewExporterWithOptions because I don't want to just break the APIs immediately, plus the direction of this PR flows in favour of replacing NewExporter entirely.

Reply:
@odeke-em @ramonza @rakyll zipkin most likely support http, kafka and scribe (but we want to run away from it). You can check [existing instrumentations](https://zipkin.io/pages/existing_instrumentations.html).

Reply:
I don't like this fact that we have two different APIs to configure the exporter now. On the other hand, NewExporter is not using the OpenCensus-native style but Zipkin constructs.

If we need to provide two ways, I'd expect NewExporter to be the primary one.

I'd not focus on breaking the API given very few users we have.